src/core/meshcop/dataset_local.cpp

```diff
@@ -193,5 +215,80 @@ Error DatasetLocal::Save(const Dataset &aDataset)
     return error;
 }
 
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+void DatasetLocal::DestroySecurelyStoredKeys() const
+{
+    using namespace Crypto::Storage;
+
+    KeyRef networkKeyRef = IsActive() ? kActiveDatasetNetworkKeyRef : kPendingDatasetNetworkKeyRef;
+    KeyRef pskcRef       = IsActive() ? kActiveDatasetPskcRef : kPendingDatasetPskcRef;
+
+    // Destroy securely stored keys associated with the given operational dataset type.
+    DestroyKey(networkKeyRef);
+    DestroyKey(pskcRef);
+}
+
+void DatasetLocal::MoveKeysToSecureStorage(Dataset &aDataset) const
+{
+    using namespace Crypto::Storage;
+
+    KeyRef         networkKeyRef = IsActive() ? kActiveDatasetNetworkKeyRef : kPendingDatasetNetworkKeyRef;
+    KeyRef         pskcRef       = IsActive() ? kActiveDatasetPskcRef : kPendingDatasetPskcRef;
+    NetworkKeyTlv *networkKeyTlv = static_cast<NetworkKeyTlv *>(aDataset.GetTlv(Tlv::kNetworkKey));
```

Comment:
How about if use this instead?  
```suggestion
    NetworkKeyTlv *networkKeyTlv = aDataset.GetTlv<NetworkKeyTlv>();
```
(also there are similar cases below in other methods which we can use the same).

- The `GetTlv<TlvType>` helps take care of `static_cast` for us.
- It is also a bit safer to use as it automatically ensures that the correct TLV type value related to `TlvType` we want is used.

Reply:
Very nice, I'll use it.